test/extended/util/client.go

```diff
@@ -323,113 +327,58 @@ func (c *CLI) setupProject() string {
+					options.FieldSelector = fieldSelector
+					return c.KubeClient().RbacV1().RoleBindings(newNamespace).List(context.Background(), options)
+				},
+				WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+					options.FieldSelector = fieldSelector
+					return c.KubeClient().RbacV1().RoleBindings(newNamespace).Watch(context.Background(), options)
+				},
 			}
-		})
-		o.Expect(err).NotTo(o.HaveOccurred())
-	}
 
-	WaitForNamespaceSCCAnnotations(c.KubeClient().CoreV1(), newNamespace)
+			_, err := watchtools.UntilWithSync(ctx, lw, &rbacv1.RoleBinding{}, nil, func(event watch.Event) (b bool, e error) {
+				switch t := event.Type; t {
+				case watch.Added, watch.Modified:
+					return true, nil
 
-	framework.Logf("Project %q has been fully provisioned.", newNamespace)
-	return newNamespace
-}
+				case watch.Deleted:
+					return true, fmt.Errorf("object has been deleted")
 
-func (c *CLI) setupNamespace() string {
-	requiresTestStart()
-	newNamespace := names.SimpleNameGenerator.GenerateName(fmt.Sprintf("e2e-test-%s-", c.kubeFramework.BaseName))
-	username := fmt.Sprintf("%s-user", newNamespace)
-	serviceAccountName := "default"
-	c.SetNamespace(newNamespace)
-
-	nsObject := &corev1.Namespace{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: newNamespace,
-			Annotations: map[string]string{
-				annotations.OpenShiftDescription: username,
-				annotations.OpenShiftDisplayName: newNamespace,
-				"openshift.io/requester":         username,
-			},
-		},
+				default:
+					return true, fmt.Errorf("internal error: unexpected event %#v", e)
+				}
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}
 	}
-	framework.Logf("Creating namespace %q", newNamespace)
-	_, err := c.AdminKubeClient().CoreV1().Namespaces().Create(context.Background(), nsObject, metav1.CreateOptions{})
-	o.Expect(err).NotTo(o.HaveOccurred())
-	c.kubeFramework.AddNamespacesToDelete(nsObject)
-
-	framework.Logf("Waiting for ServiceAccount %q to be provisioned...", serviceAccountName)
-	err = WaitForServiceAccount(c.AdminKubeClient().CoreV1().ServiceAccounts(newNamespace), serviceAccountName)
-	o.Expect(err).NotTo(o.HaveOccurred())
 
-	framework.Logf("Configuring kubeconfig with user %q certificates...", username)
-	c.ChangeUser(username)
-
-	framework.Logf("Waiting for RoleBinding %q to be provisioned...", username)
-	err = c.setupRoleInNamespace(username)
+	err = WaitForNamespaceSCCAnnotations(c.KubeClient().CoreV1(), newNamespace)
 	o.Expect(err).NotTo(o.HaveOccurred())
 
-	framework.Logf("Waiting on permissions in namespace %q ...", newNamespace)
-	err = WaitForSelfSAR(1*time.Second, 60*time.Second, c.KubeClient(), kubeauthorizationv1.SelfSubjectAccessReviewSpec{
-		ResourceAttributes: &kubeauthorizationv1.ResourceAttributes{
-			Namespace: newNamespace,
-			Verb:      "create",
-			Group:     "",
-			Resource:  "pods",
-		},
-	})
-	o.Expect(err).NotTo(o.HaveOccurred())
-
-	err = c.setupNamespacePodSecurity(newNamespace)
```

Comment:
Why we don't need this anymore?. Because we are already setting it to `restricted` in somewhere?

Reply:
I'm squashing two methods into one, to ensure we always use the same code path, and clean all the resources after the tests.